dbt/include/databricks/macros/materializations/incremental/strategies.sql

```diff
@@ -51,24 +51,32 @@
 {% macro get_insert_into_sql(source_relation, target_relation) %}
 
     {%- set common_columns = [] -%}
-    {%- set source_columns = adapter.get_columns_in_relation(source_relation) | map(attribute="name") -%}
-    {%- for dest_col in adapter.get_columns_in_relation(target_relation) -%}
-      {% if dest_col.name in source_columns %}
-          {%- if common_columns.append(dest_col) -%}{%- endif -%}
+    {%- set source_columns = adapter.get_columns_in_relation(source_relation) | map(attribute="quoted") | list -%}
+    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) | map(attribute="quoted") | list -%}
+    {{ log("source_columns: " ~ (source_columns | join(', '))) }}
+    {{ log("dest_columns: " ~ (dest_columns | join(', '))) }}
+    {%- for dest_col in dest_columns -%}
+      {%- if dest_col in source_columns %}
```

Comment:
so if source has a column not in destination, what do we do?

Reply:
source column will be in destination unless the setting is ignore (because the schema is altered to include them before this called).  If setting is ignore, we are intentionally not taking source columns that are not in destination.